nakadi-producer/src/main/java/org/zalando/nakadiproducer/eventlog/EventLogWriter.java

```diff
@@ -42,6 +44,24 @@
     @Transactional
     void fireCreateEvent(String eventType, String dataType, Object data);
 
+  /**
+   * Fires data change events about the creation of some resources (objects), see
+   * {@link #fireCreateEvent(String, String, Object) fireCreateEvent} for more details.
+   *
+   * @param eventType
+   *            the Nakadi event type of the event. This is roughly equivalent
+   *            to an event channel or topic.
+   *
+   * @param dataTypeToData
+   *            the content of the {@code data_type} field of the Nakadi
+   *            event mapped to some POJOs that can be serialized into JSON (required
+   *            parameter). This is meant to be a representation of the
+   *            current state of the resource. It will be used as content of
+   *            the {@code data} field of the Nakadi event.
+   */
+    @Transactional
+    void fireCreateEvents(String eventType, Map<String, Collection<Object>> dataTypeToData);
```

Comment:
I guess I missed this on my first round (or these methods were not yet there?).
Is there a reason we pass a map here instead of just one data type and a collection of DTOs?
For all data event types I know, the `data_type` attribute is effectively constant (and/or ignored).

Do you have examples where regularly different values of `data_type` are sent on the same event type?
If not, I would go for the simpler signature:

```suggestion
   * @param dataType
   *            the content of the {@code data_type} field of the Nakadi
   *            event
   * @param data
   *            some POJOs that can be serialized into JSON (required
   *            parameter). This is meant to be a representation of the
   *            current state of the resource. It will be used as content of
   *            the {@code data} field of the Nakadi event.
   */
    @Transactional
    void fireCreateEvents(String eventType, String dataType, Collection<Object> data);
```

Reply:
Could you do the same change for the other `fire` methods?